app/controllers/incomeSources/add/CheckBusinessDetailsController.scala

```diff
@@ -180,4 +175,50 @@ class CheckBusinessDetailsController @Inject()(val checkBusinessDetails: CheckBu
         }
   }
 
+  private def getBusinessDetailsFromSession(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, CheckBusinessDetailsViewModel]] = {
+    val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
+    val skipAccountingMethod: Boolean = userActiveBusinesses.isEmpty
+
+    val result = for {
+      businessName <- sessionService.get(SessionKeys.businessName)
```

Comment:
this seems quite strange approach as you keep querying SessionService regardless there was an error or not,
if will be fine to return first error occurred ; you can re-use foldLeft logic we used in this PR already